sklearn/utils/validation.py

```diff
@@ -506,14 +507,38 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning)
+
+    # Indices Datatype regulation
+    _check_large_sparse(array, accept_large_sparse)
+
     return array
 
 
+def _check_large_sparse(array, accept_large_sparse=False):
+    """
+    Indices Regulation for CSR Matrices.
+    Only Int32 are supported for now
+
+    """
+    if accept_large_sparse is False and type(array) == sp.csr.csr_matrix:
+        supported_indices = ["int32"]
+        if (hasattr(array, "indices") and hasattr(array.indices, "dtype") and
+                array.indices.dtype not in supported_indices):
+            raise ValueError("CSR arrays accepts only 32 bit integer indexing."
+                             " You provided % s bit indexing"
+                             % array.indices.dtype)
+        if (hasattr(array, "indptr") and hasattr(array.indptr,  "dtype") and
```

Comment:
what case does this handle? It's not tested.

Reply:
basically if any of indices or indptr are 64bit,
sparse will crash (so to check OR condition)